apps/hash-ai-worker-ts/src/activities/infer-entities/shared/generate-propose-entities-tools.ts

```diff
@@ -26,15 +27,59 @@ export type ProposedEntityToolCreationsByType = Record<
   ProposedEntityWithSimplifiedProperties[]
 >;
 
-export const generateProposeEntitiesTools = (
+export const generateToolLinkFields = (params: {
+  canLinkToExistingEntities?: boolean;
+}) => ({
+  sourceEntityId: {
+    description: dedent(`
+      The entityId of the source entity of the link.
+      ${params.canLinkToExistingEntities ? "If the source is an existing entity, it must be a string. If it is a proposed entity, it must be a number." : ""}
+    `),
+    ...(params.canLinkToExistingEntities
+      ? {
+          oneOf: [
+            {
+              type: "number",
+            },
+            {
+              type: "string",
+            },
+          ],
+        }
+      : { type: "number" }),
```

Comment:
I wonder if the browser plugin breaks if we just make these all strings... possibly not as it's still building off this PR, where the types are now either a string or a number.

Reply:
> once we unify the ProposedEntity types it would cause type errors there, and probably some other places (like in the inferEntities activity).

Yes, fair enough. I think these will be minimal as the only real difference is the type of `entityId` (and also now the shape of linked entity ids), but all easily fixed.